cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -369,13 +369,16 @@ impl Inst {
     }
 
     /// Emits zkASM to trace executed instructions.
+    ///
+    /// Calls a method implemented by the `InstructionTracer.js` helper. An instance of this helper
+    /// must be provided to `zkevm-proverjs` for the successful execution of the generated zkASM.
     fn emit_inst_instrumentation(&self, sink: &mut MachBuffer<Inst>) {
         match self {
             // Labels are handled separately since benchmarking will provide a separate command to
             // analyze labels.
             &MInst::Label { .. } => {}
             _ => put_string(
-                ";; TODO(mooori) call the helper to trace `MInst` execution\n",
+                &format!("$${{traceInstruction({})}}\n", self.print_name()),
```

Comment:
Just a brain dump, nothing actionable for this PR:

IF we wanted this instrumentation to be performance-sensitive, instrumenting *just* around the labels would be the ideal thing to do. After all, the instructions executed in between any two labels can be recovered entirely by looking at the zkwasm code. Doing so would reduce the overhead of the instrumentation appreciably.

Building tooling like that sounds non-trivial however, nor do we care about instrumented code performance (do we?) so I think its perfectly fine to do what you've done.

Reply:
My understanding is that for the time being we don't really care about instrumented code performance and the programs that are profiled terminate relatively quickly (like the sha256 and Keccak benchmarks).

Based on the functionality added for instruction tracing it should be relatively simple to do the same for labels. I would be happy to implement it for labels if we have capacity or if it becomes neccessary in the future due to performance reasons.